third_party/move/move-vm/runtime/src/loader/function.rs

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    loader::{Loader, Module, Resolver, ScriptHash},
+    native_functions::{NativeFunction, NativeFunctions, UnboxedNativeFunction},
+};
+use move_binary_format::{
+    access::ModuleAccess,
+    errors::{PartialVMError, PartialVMResult},
+    file_format::{
+        AbilitySet, Bytecode, CompiledModule, FunctionDefinition, FunctionDefinitionIndex,
+        Visibility,
+    },
+};
+use move_core_types::{identifier::Identifier, language_storage::ModuleId, vm_status::StatusCode};
+use move_vm_types::loaded_data::runtime_types::Type;
+use std::{fmt::Debug, sync::Arc};
+
+// A simple wrapper for the "owner" of the function (Module or Script)
+#[derive(Debug)]
+pub(crate) enum Scope {
+    Module(ModuleId),
+    Script(ScriptHash),
+}
+
+// A runtime function
+// #[derive(Debug)]
+// https://github.com/rust-lang/rust/issues/70263
+pub(crate) struct Function {
+    #[allow(unused)]
+    pub(crate) file_format_version: u32,
+    pub(crate) index: FunctionDefinitionIndex,
+    pub(crate) code: Vec<Bytecode>,
+    pub(crate) type_parameters: Vec<AbilitySet>,
+    pub(crate) native: Option<NativeFunction>,
```

Comment:
And when it is native, the bytecode is empty? I would introduce an enum then, and you can skip `def_is_native`.